manifests/init.pp

```diff
@@ -160,86 +123,81 @@
+#   Defaults to the result of a PuppetDB query, or `[$trusted['certname']]`
 #
-# @example Enable Telegraf, Graphite, and Archive
-#  class { 'puppet_metrics_dashboard':
-#    add_dashboard_examples => true,
-#    influxdb_database_name => ['puppet_metrics','telegraf','graphite'],
-#    consume_graphite       => true,
-#    configure_telegraf     => true,
-#    enable_telegraf        => true,
-#  }
+# @param postgres_host_list
+#   An Array of servers that Telegraf will collect PostgreSQL metrics from. 
+#   Entries may be:
+#     - A String that contains a hostname or IP address.
+#       (The module will use a default port number of `5432`)
+#     - An Array where the first entry is a String that contains a hostname or IP address,
+#       and the second entry is an Integer that specifies the port number.
+#   Defaults to the results of a PuppetDB query, or `[$trusted['certname']]`
 #
-# @example Enable SSL
-#   class { 'puppet_metrics_dashboard':
-#     use_dashboard_ssl => true,
-#   }
+# @param puppetdb_metrics
+#   An Array of Hashes containing name/url pairs for each PuppetDB metric.
+#   Refer to `functions/puppetdb_metrics.pp` for defaults.
 #
 class puppet_metrics_dashboard (
-  Boolean $add_dashboard_examples,
   Boolean $manage_repos,
-  Boolean $use_dashboard_ssl,
-  String $dashboard_cert_file,
-  String $dashboard_cert_key,
+
+  Boolean $add_dashboard_examples,
   Boolean $overwrite_dashboards,
-  String $overwrite_dashboards_file,
-  String $influx_db_service_name,
-  Array[String] $influxdb_database_name,
-  String $grafana_version,
-  Integer $grafana_http_port,
-  String $influx_db_password,
-  String $grafana_password,
-  Boolean $enable_kapacitor,
+
   Boolean $enable_chronograf,
+  Boolean $enable_kapacitor,
   Boolean $enable_telegraf,
   Boolean $configure_telegraf,
   Boolean $consume_graphite,
-  Puppet_metrics_dashboard::HostList $master_list,
-  Puppet_metrics_dashboard::HostList $puppetdb_list,
-  Puppet_metrics_dashboard::HostList $postgres_host_list,
+
+  Array[String] $influxdb_database_name,
   Array[String] $influxdb_urls,
+
+  String $influx_db_service_name,
+  String $influx_db_password,
+
   String $telegraf_db_name,
   String[2] $telegraf_agent_interval,
   String[2] $http_response_timeout,
   String[2] $pg_query_interval,
+
+  Boolean $use_dashboard_ssl,
+  String $dashboard_cert_file,
+  String $dashboard_cert_key,
+
+  Integer $grafana_http_port,
+  String $grafana_password,
+  String $grafana_version,
+
+  String $overwrite_dashboards_file,
+
   Hash $grafana_config,
+
+  Puppet_metrics_dashboard::HostList $master_list        = puppet_metrics_dashboard::localhost_or_hosts_with_pe_profile('master'),
+  Puppet_metrics_dashboard::HostList $puppetdb_list      = puppet_metrics_dashboard::localhost_or_hosts_with_pe_profile('puppetdb'),
+  Puppet_metrics_dashboard::HostList $postgres_host_list = puppet_metrics_dashboard::localhost_or_hosts_with_pe_profile('database'),
+
```

Comment:
The above are the functional changes to this manifest: querying PuppetDB before defaulting to `$trusted['certname']`.